vllm/model_executor/layers/fused_moe/flashinfer_cutlass_moe.py

```diff
@@ -170,8 +170,6 @@ def apply(
             "w1_scale and w2_scale must not "
             "be None for FlashInferExperts")
 
-        assert not apply_router_weight_on_input
```

Comment:
It looks like apply_router_weight_on_input isn't being used at all in this function, how does the moe know to apply this?

Reply:
I have added it in the "prepare" part and confirmed that it improved the accuracy